src/ck-core/charm.h

```diff
@@ -189,6 +189,14 @@ typedef enum{
 	TypeArray
 } ChareType;
 
+// What to do if an entry method is invoked on
+// an array element that does not (yet) exist:
+typedef enum{
+	CkArray_IfNotThere_buffer = 0,			// Wait for it to be created
+	CkArray_IfNotThere_createhere = 1,	// Make it on sending Pe
+	CkArray_IfNotThere_createhome = 2		// Make it on (a) home Pe
+} CkArray_IfNotThere;
```

Comment:
This should preserve the type that was given in the old version and should be formatted to match the surrounding context. Also, maybe make this into an `enum class`?

Reply:
What would you like me to change formatting-wise?

Unfortunately, the type cannot be preserved nor can this be made an `enum class` since it's all inside an `extern "C"` block. Since `CkArray_IfNotThere` should never be sent via a message, the underlying type may not matter?

Reply:
'ifNotThere' actually has to be sent in chare array msgs so that the runtime knows what to do with messages for chare array elements that aren't created yet. It is defined as part of the message envelope as a 'uchar' in src/ck-core/envelope.h's u_type union. It looks like we're already implicitly casting from int to uchar inside envelope::setArrayIfNotThere() though

Reply:
Yes, the type-cast of the policy information to put it in the `envelope` should be safe-ish here.

Alternatively, I can move the declaration to `charm++.h`, whose contents are not confined to an `extern "C"` block.